File: app/src/main/java/com/example/calculator/MainActivity.java
package com.example.calculator;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    TextView result;
    String currentInput = "";
    String operator = "";
    double firstNumber = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        result = findViewById(R.id.result);
    }

    public void onNumberClick(View view) {
        Button button = (Button) view;
        currentInput += button.getText().toString();
        result.setText(currentInput);
    }

    public void onOperatorClick(View view) {
        Button button = (Button) view;
        firstNumber = Double.parseDouble(currentInput);
        operator = button.getText().toString();
        currentInput = "";
    }

    public void onEqualClick(View view) {
        double secondNumber = Double.parseDouble(currentInput);
        double finalResult = 0;

        switch (operator) {
            case "+":
                finalResult = firstNumber + secondNumber;
                break;
            case "-":
                finalResult = firstNumber - secondNumber;
                break;
            case "×":
                finalResult = firstNumber * secondNumber;
                break;
            case "÷":
                finalResult = firstNumber / secondNumber;
                break;
        }

        result.setText(String.valueOf(finalResult));
        currentInput = String.valueOf(finalResult);
    }

    public void onClearClick(View view) {
        currentInput = "";
        operator = "";
        firstNumber = 0;
        result.setText("0");
    }
}

File: app/src/main/java/com/example/linkdownloader/MainActivity.kt
package com.example.linkdownloader

import android.app.DownloadManager
import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.os.Environment
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val urlInput = findViewById<EditText>(R.id.urlInput)
        val downloadButton = findViewById<Button>(R.id.downloadButton)

        downloadButton.setOnClickListener {
            val url = urlInput.text.toString().trim()

            if (url.isEmpty()) {
                Toast.makeText(this, "أدخل الرابط أولاً", Toast.LENGTH_SHORT).show()
            } else {
                downloadFile(url)
            }
        }
    }

    private fun downloadFile(url: String) {
        try {
            val request = DownloadManager.Request(Uri.parse(url))
            request.setTitle("Downloading file")
            request.setDescription("Downloading from link")
            request.setNotificationVisibility(
                DownloadManager.Request.VISIBILITY_VISIBLE_NOTIFY_COMPLETED
            )
            request.setDestinationInExternalPublicDir(
                Environment.DIRECTORY_DOWNLOADS,
                Uri.parse(url).lastPathSegment
            )

            val manager = getSystemService(Context.DOWNLOAD_SERVICE) as DownloadManager
            manager.enqueue(request)

            Toast.makeText(this, "بدأ التحميل...", Toast.LENGTH_SHORT).show()

        } catch (e: Exception) {
            Toast.makeText(this, "رابط غير صالح", Toast.LENGTH_SHORT).show()
        }
    }
}